test/metabase/query_analysis_test.clj

```diff
@@ -10,23 +10,64 @@
    [metabase.test :as mt]
    [toucan2.tools.with-temp :as t2.with-temp]))
 
-(deftest native-query-enabled-test
-  (mt/discard-setting-changes [sql-parsing-enabled]
-    (testing "sql parsing enabled"
+(deftest legacy-sql-parsing-enabled-test
+  (mt/discard-setting-changes [query-analysis-native-disabled sql-parsing-enabled]
+    (testing "legacy setting enabled"
       (public-settings/sql-parsing-enabled! true)
-      (binding [query-analysis/*parse-queries-in-test?* true]
-        (is (true? (query-analysis/enabled? :native)))))
-    (testing "sql parsing disabled"
+      ;; I'm not sure why this is nil, but it is fine for our purposes.
+      (is (nil? (public-settings/query-analysis-native-disabled)))
```

Comment:
this is worth chasing down, I think. I suspect that `init` isn't doing what we hope? Do we need a `default`?

Reply:
We can't mix `init` and `default`, I tried that first - and got lectured by my own sanity check code 😆

Reply:
I'm pretty curious what's going on, so I'll put a bit of time in tomorrow debugging.